A59-audit-logging.md

```diff
@@ -0,0 +1,349 @@
+address this issue, the rationale of which will be elaborated more in
+[one of the design alternatives](#audit-condition-in-the-http-connection-manager).
+
+### gRPC Authorization Policy Changes
+
+The gRPC authorization policy consists of `deny_rules` and `allow_rules`, which
+are mapped to two RBACs. Overall it's a subset of the RBAC as stated in [A43: gRPC authorization API][A43].
+We will keep this philosophy by adding just one `audit_logging_options` in the
+policy instead of having separate ones for those two sets of rules.
+
+Following are changes to the JSON schema of gRPC Authorization Policy introduced
+in [A43: gRPC authorization API][A43]:
+
+```json
+{
+  "title": "AuthorizationPolicy",
+  "definitions": {
+    ...
+    "audit_logger": {
+      "description": "Configuration for the audit logger.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The name of the audit logger configuration."
+            "It is mainly used for human purposes.",
+          "type": "string",
+        },
+        "typed_config": {
+          "description": "The typed config for the audit logger."
+            "This needs to be a json object mapped from google.protobuf.Struct"
+            "proto message.",
+          "type": "object",
+        }
+      },
+    }
+  },
+  "properties": {
+    ...
+    "audit_logging_options": {
+      "description": "Audit logging options that include the condition for audit"
+        "logging to happen and audit logger configurations.",
+      "type": "object",
+      "properties": {
+        "audit_condition": {
+          "description": "Condition for the audit logging to happen."
+            "Same as NONE if ommited.",
+          "type": "string",
+          "enum": ["NONE", "ON_DENY", "ON_ALLOW", "ON_DENY_AND_ALLOW"]
+        },
+        "audit_logger": {
+          "description": "Audit logger configurations",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/audit_logger"
+          }
+        }
+      }
+    },
+  },
+  "required": ["name", "allow_rules"]
+}
+```
+
+For readers that are more familiar with proto messages, the change translates
+to:
+
+```
+message AuthorizationPolicy {
+  …
+
+  message AuditLoggingOptions {
+    enum AuditCondition {
+      NONE = 0;
+      ON_DENY = 1;
+      ON_ALLOW = 2;
+      ON_DENY_AND_ALLOW = 3;
+    }
+
+    AuditCondition audit_condition = 1;
+
+    message AuditLogger {
+      string name = 1;
+      google.protobuf.Struct config = 2;  
+    }
+    
+    repeated AuditLogger audit_loggers = 2;
+  }
+
+  AuditLoggingOptions audit_logging_options = 4;
+}
+```
+
+Note that the definition above is only for illustration purposes and does not
+actually exist anywhere as a `.proto` file. Nor does gRPC process the policy
+as protobuf by any means.
+
+The authorization policy is backed by two RBAC filters, so the audit logger
```

Comment:
You say "RBAC with DENY" later, but don't really introduce it. Somewhere here, where we introduce "two RBAC filters" it may be good to say that one is DENY and one is ALLOW.

Maybe:
> The authorization policy is backed by two RBAC filters, a DENY followed by an ALLOW. The audit logger...